spinalcordtoolbox/utils/fs.py

```diff
@@ -264,22 +263,111 @@ def relpath_or_abspath(child_path, parent_path):
         return abspath
 
 
-@contextmanager
-def mutex(name):
+class Mutex(portalocker.BoundedSemaphore):
+    """
+    General purpose mutex (mutually exclusive semaphore) with a tweaks to
+    make it safer, more consistent, and easier to use within SCT. Namely:
+
+    * Implicitly sanitizes the key used by the Mutex to prevent unintended side effects
+    * Its "maximum" is enforced to be 1, as required of a mutex.
+    * Can be used as a context manager, ensuring it is released after use.
+    * If provided, can log a message when it starts waiting to acquire its lock.
+
+    Use instances of this class to ensure that processes run in parallel will not
+    create race conditions when they try to access something another process is currently
+    modifying (i.e. when writing to a shared log/QC directory).
+
+    :param key: The key this Mutex will check against. If multiple mutexes share the same key,
+        they will share access rights, even if run in different processes on the same machine.
+    :param directory: Where the files used to track the mutex will be placed.
+    :param timeout: How long (in seconds) the Mutex should wait when trying to acquire its corresponding
+        lock before failing.
+    :param check_interval: How often (in seconds) the Mutex should try to acquire the lock while it waits.
+    :param fail_when_locked: Whether to throw an exception if the Mutex fails to acquire the lock within
+        the timeout period.
+    :param waiting_msg: Message to display when waiting to acquire the lock. If None, no message will print.
     """
-    Use a bounded semaphore as a mutex to prevent parallel processes from running.
+    def __init__(
+        self,
+        key: str,
+        directory: str = tempfile.gettempdir(),
+        timeout: float | None = 5,  # 5 seconds
+        check_interval: float | None = 0.25,  # Every quarter second
+        fail_when_locked: bool | None = True,
```

Comment:
Instead of hardcoding the defaults of (the current version of) portalocker's `BoundedSemaphore`, I would suggest using `**kwargs` and passing them on to `super().__init__` (if we want to consider `Mutex` to be a thin wrapper).

If we want to consider `Mutex` to be a self-contained abstraction instead, I would suggest hiding anything we don't actually use from `BoundedSemaphore` as an implementation detail.

(These suggestions are coming from a principle of trying to have a single source of truth, because duplicated truths tend to get out of sync. But, I can see an argument that it's very convenient to have local documentation, instead of having to dig around a dependency's source code for details.)